specs/phase1/validator.md

```diff
@@ -0,0 +1,529 @@
+
+- Let `light_client_committee = get_light_client_committee(state, compute_epoch_at_slot(slot))`
+
+##### Light client vote data
+
+First the validator constructs `light_client_vote_data`, a [`LightClientVoteData`](#lightclientvotedata) object.
+
+* Let `head_block` be the result of running the fork choice during the assigned slot.
+* Set `light_client_vote.slot = slot`.
+* Set `light_client_vote.beacon_block_root = hash_tree_root(head_block)`.
+
+###### `LightClientVoteData`
+
+```python
+class LightClientVoteData(Container):
+    slot: Slot
+    beacon_block_root: Root
+```
+
+##### Construct vote
+
+Then the validator constructs `light_client_vote`, a [`LightClientVote`](#lightclientvote) object.
+
+* Set `light_client_vote.data = light_client_vote_data`.
+* Set `light_client_vote.aggregation_bits` to be a `Bitvector[LIGHT_CLIENT_COMMITTEE_SIZE]`, where the bit of the index of the validator in the `light_client_committee` is set to `0b1` and all other bits are are set to `0b0`.
+* Set `light_client_vote.signature = vote_signature` where `vote_signature` is obtained from:
+
+```python
+def get_light_client_vote_signature(state: BeaconState,
+                                    light_client_vote_data: LightClientVoteData,
+                                    privkey: int) -> BLSSignature:
+    domain = get_domain(state, DOMAIN_LIGHT_CLIENT, compute_epoch_at_slot(light_client_vote_data.slot))
+    signing_root = compute_signing_root(light_client_vote_data, domain)
+    return bls.Sign(privkey, signing_root)
+```
+
+###### `LightClientVote`
+
+```python
+class LightClientVote(Container):
+    data: LightClientVoteData
+    aggregation_bits: Bitvector[LIGHT_CLIENT_COMMITTEE_SIZE]
+    signature: BLSSignature
+```
+
+##### Broadcast
+
+Finally, the validator broadcasts `light_client_vote` to the `light_client_votes` pubsub topic.
+
+#### Light client vote aggregation
+
+Some validators in the light client committee are selected to locally aggregate light client votes with a similar `light_client_vote_data` to their constructed `light_client_vote` for the assigned `slot`.
+
+#### Aggregation selection
+
+A validator is selected to aggregate based upon the return value of `is_light_client_aggregator()`.
+
+```python
+def get_light_client_slot_signature(state: BeaconState, slot: Slot, privkey: int) -> BLSSignature:
+    domain = get_domain(state, DOMAIN_LIGHT_SELECTION_PROOF, compute_epoch_at_slot(slot))
+    signing_root = compute_signing_root(slot, domain)
+    return bls.Sign(privkey, signing_root)
+```
+
+```python
+def is_light_client_aggregator(state: BeaconState, slot: Slot, slot_signature: BLSSignature) -> bool:
+    committee = get_light_client_committee(state, compute_epoch_at_slot(slot))
+    modulo = max(1, len(committee) // TARGET_LIGHT_CLIENT_AGGREGATORS_PER_SLOT)
```

Comment:
Note that compare to `is_aggregator`, it's highly possible that `len(committee) == LIGHT_CLIENT_COMMITTEE_SIZE` and then `module == 16`. So the `is_light_client_aggregator` output would be more predictable.